### PR TITLE
add Infinity gloabal property (#480)

### LINKED ISSUE #480: Implement global `Infinity` property
**ECMASCript feature**

`Infinity` is not a number, but a global property.

We can add this by adding a new module like `NaN` https://github.com/boa-dev/boa/tree/master/boa/src/builtins/nan and then adding it to the builtin `init` function in `builtins/mod.rs` the value it should contain should be `Value::from(f64::INFINITY)`

More information
	- [ECMAScript reference][spec]
    - [MDN][mdn]

[spec]: https://tc39.es/ecma262/#sec-value-properties-of-the-global-object-infinity
[mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Infinity

### PR DESCRIPTION
**Pull request by AnirudhKonduru**
This Pull Request fixes/closes #480.

It changes the following:
 - adds the Infinity global property

### CODE REVIEW

File: boa/src/builtins/infinity/mod.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+//! This module implements the global `Infinity` property.
+//!
+//! The global `Infinity` is a property of the global object. In other words,
+//! it is a variable in global scope.
```
Comment (HalidOdat) [inline boa/src/builtins/infinity/mod.rs:4]: A better doc would be:
```suggestion
//! The global property `Infinity` is a numeric value representing infinity.
```

File: boa/src/builtins/infinity/tests.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+use crate::exec;
+
+#[test]
```
Comment (HalidOdat) [inline boa/src/builtins/infinity/tests.rs:3]: Another test would be to test if `Number.POSITIVE_INFINITY === Infinity` should evaluate to `true`

Comment (AnirudhKonduru) [reply boa/src/builtins/infinity/tests.rs:3]: thanks! I've added 
`
#[test]
fn infinity_exists_and_equals_to_number_positive_infinity_value() {
    let scenario = r#"
        Number.POSITIVE_INFINITY === Infinity;
        "#;

    assert_eq!(&exec(scenario), "true");
}`

looks fine?

File: boa/src/builtins/infinity/mod.rs (hunk 1 above)
Comment (AnirudhKonduru) [reply boa/src/builtins/infinity/mod.rs:4]: fixed it, thanks!

File: boa/src/builtins/infinity/tests.rs (hunk 2 above)
Comment (HalidOdat) [reply boa/src/builtins/infinity/tests.rs:3]: > looks fine?

Yes. Looks good :)

**Comment by AnirudhKonduru** [comment]
my first PR 😄 @HalidOdat thank you! the info on #480 made it super easy.

**Comment by HalidOdat** [comment]
Lets merge this :)

**Pull request merged**